packages/typescript-estree/src/convert.ts

```diff
@@ -643,6 +643,19 @@ export class Converter {
     return this.converter(child, parent, false);
   }
 
+  /**
+   * Converts TypeScript node array into an ESTree node list.
+   * @param children the child `ts.NodeArray` or `ts.Node[]`
+   * @param parent parentNode
+   * @returns the converted ESTree node list
+   */
+  private convertChildren(
+    children?: ts.Node[] | ts.NodeArray<ts.Node>,
```

Comment:
This is never invoked with the `children` parameter `undefined`... So let's either require it

```suggestion
    children: ts.Node[] | ts.NodeArray<ts.Node>,
```

or add logic to handle the undefined array the way the call sites do? Like `return (children ?? []).map(...)`